src/thunderbots/software/ai/hl/stp/evaluation/indirect_chip.cpp

```diff
@@ -0,0 +1,269 @@
+{
+    Point p1 = triangle[0];
+    Point p2 = triangle[1];
+    Point p3 = triangle[2];
+
+    double center_x = (p1.x() + p2.x() + p3.x()) / 3;
+    double center_y = (p1.y() + p2.y() + p3.y()) / 3;
+
+    Point center = Point(center_x, center_y);
+
+    return center;
+}
+
+double Evaluation::getTriangleArea(Triangle triangle)
+{
+    Point p1 = triangle[0];
+    Point p2 = triangle[1];
+    Point p3 = triangle[2];
+
+    double area = abs(0.5 * ((p2.x() - p1.x()) * (p3.y() - p1.y()) -
+                             (p3.x() - p1.x()) * (p2.y() - p1.y())));
+
+    return area;
+}
+
+std::vector<Triangle> Evaluation::remove_outofbounds_triangles(
+    const World &world, std::vector<Triangle> triangles)
+{
+    std::vector<Triangle> valid_triangles;
+    std::vector<Point> chip_area_corners = findBestChipTargetArea(
+        world, Util::DynamicParameters::Evaluation::Indirect_Chip::chip_target_area_inset
+                   .value());
+    Point center;
+
+    double smallest_x = chip_area_corners[0].x();
+    double smallest_y = chip_area_corners[0].y();
+    double largest_x  = chip_area_corners[0].x();
+    double largest_y  = chip_area_corners[0].y();
+
+    for (unsigned int i = 0; i < chip_area_corners.size(); i++)
+    {
+        Point p = chip_area_corners[i];
+        if (p.x() < smallest_x)
+            smallest_x = p.x();
+        if (p.x() > largest_x)
+            largest_x = p.x();
+        if (p.y() < smallest_y)
+            smallest_y = p.y();
+        if (p.y() > largest_y)
+            largest_y = p.y();
+    }
+
+    for (unsigned int i = 0; i < triangles.size(); i++)
+    {
+        Triangle t = triangles[i];
+        center     = getTriangleCenter(triangles[i]);
+        if (center.x() <= largest_x && center.x() >= smallest_x &&
+            center.y() <= largest_y && center.y() >= smallest_y)
+        {
+            valid_triangles.push_back(t);
+        }
+    }
+
+    return valid_triangles;
+}
+
+Rectangle Evaluation::findBestChipTargetArea(const World &world, double inset)
+{
+    double ballX     = world.ball().position().x();
+    double fieldX    = world.field().enemyGoal().x() - inset;
+    double negFieldY = world.field().enemyCornerNeg().y() + inset;
+    double posFieldY = world.field().enemyCornerPos().y() - inset;
+
+    std::vector<Point> corners = {Point(ballX, negFieldY), Point(ballX, posFieldY),
+                                  Point(fieldX, negFieldY), Point(fieldX, posFieldY)};
+
+    Rectangle target_rectangle = Rectangle(Point(ballX, negFieldY), Point(fieldX, posFieldY));
+    Point p1 = r[0]; // south-west corner
+    Point p2 = r[1]; // north-west corner
+    Point p3 = r[2]; // north-east corner
+    Point p4 = r[3]; // south-east corner
+
+    return target_rectangle;
+}
+
+std::optional<Triangle> Evaluation::getLargestValidTriangle(
+    std::vector<TrianSgle> allTriangles, double min_area, double min_edge_len,
+    double min_edge_angle)
+{
+    Triangle largest    = allTriangles[0];
```

Comment:
Try adding a test for this function where you give it an empty vector of triangles. I think this line might cause some issues.

Reply:
I spot this... interesting typo: `std::vector<TrianSgle>`... 
I think `catkin_make` isn't complaining anymore about this line specifically. Added a test with an empty vector triangle as input, still have yet to run it though because other stuff isn't compiling yet.

Reply:
Funky.... I just saw that, I'll be pretty shocked if that doesn't break compilation.